docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,187 @@
+== oVirt OR RHV) then register the oVirt provider.
+
+##### Step 5: Add calls to the provider functions
+
+Whenever we need to call some provider-specific function we should use the provider interface instead of a switch case.
+
+##### Step 6: Implement Default Provider
+
+Will contain default implementation of most functions required by Provider interface. 
+It should be embedded in all providers to make adding new APIs easier.
+
+##### Step 7: Implement the oVirt provider and migrate Baremetal and vSphere providers
+
+Implement the oVirt provider and migrate existing providers to the Provider interface.
+
+#### Changes to the assisted-agent project:
+
+After the agent is running on the host, the provider should be determined from the OS details of the node.
+The [GetVendor](https://github.com/openshift/assisted-installer-agent/blob/master/src/inventory/system_vendor.go#L29-L45)
+method calculates if a node is virtual or not and populates the SystemVendor model which is later being used by the
+assisted service to determine if a node is running on a certain provider.
+We need to make sure that the provider is listed in the 
+[isVirtual](https://github.com/openshift/assisted-installer-agent/blob/master/src/inventory/system_vendor.go#L15-L19]
+list and that it can be detected using the fields in 
+[SystemVendor](https://github.com/openshift/assisted-service/blob/master/models/system_vendor.go).
+
+#### Changes to the Openshift installer project:
+
+A small change is required in the installer project.
+Since the openshift installer is generating a random InfraID the names of the Nodes(in the manifest) are different
+from the nodes the user provisions, and since we have a provider set then the provider will try to start new machines 
+due to the naming mismatch.
+Since the InfraID is random we can't tell the user to create the Nodes with specific names.
+Also, the InfraID is later used by the platforms in various ways to identify the cluster,
+for example in oVirt we use it to set an oVirt tag which helps us group the cluster resources in oVirt,
+and we rely on it in various places such as cluster destroy or CSI/Machine provider logic to filter cluster resources
+quickly and prevent unnecessary API calls.
+
+We need to add an env var to allow us to override the generated InfraID.
+We started a PR on the [installer project](https://github.com/openshift/installer/pull/5101).
+
+### Risks and Mitigations
+
+As with every large code change, this change contains the risk of breaking existing functionality.
+This can be mitigated by sanity testing and making sure each provider implements unit tests and e2e test suite.
+
+## Design Details
+
+### Open Questions
+
+1. Throught the assisted installer code we have structs and functions that are specific to baremetal(at least from the name),
+- Are they safe to be extended to support providers?
```

Comment:
The platform struct is already extended by the `None` and `Vsphere` platform

Reply:
yes but there are many of them not only the install config...
I think this is more of a general question to keep in mind as we develop, plus maybe it will make sense to edit some of them to have a general name that is not [SOMETHING]baremetal... not a must but will probably make sense in some places